k8s/src/main/scala/io/buoyant/k8s/v1.scala

```diff
@@ -15,33 +15,31 @@ package object v1 {
 
   @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
   @JsonSubTypes(Array(
-    new JsonSubTypes.Type(value = classOf[EndpointsWatch.Added], name = "ADDED"),
-    new JsonSubTypes.Type(value = classOf[EndpointsWatch.Modified], name = "MODIFIED"),
-    new JsonSubTypes.Type(value = classOf[EndpointsWatch.Deleted], name = "DELETED"),
-    new JsonSubTypes.Type(value = classOf[EndpointsWatch.Error], name = "ERROR")
+    new JsonSubTypes.Type(value = classOf[Added], name = "ADDED"),
+    new JsonSubTypes.Type(value = classOf[Modified], name = "MODIFIED"),
+    new JsonSubTypes.Type(value = classOf[Deleted], name = "DELETED"),
+    new JsonSubTypes.Type(value = classOf[Error], name = "ERROR")
   ))
   sealed trait EndpointsWatch extends Watch[Endpoints]
-  object EndpointsWatch {
-    case class Added(
-      `object`: Endpoints
-    ) extends EndpointsWatch with Watch.Added[Endpoints]
-
-    case class Modified(
-      `object`: Endpoints
-    ) extends EndpointsWatch with Watch.Modified[Endpoints]
-
-    case class Deleted(
-      `object`: Endpoints
-    ) extends EndpointsWatch with Watch.Deleted[Endpoints]
-
-    case class Error(
-      @JsonProperty(value = "object") status: Status
-    ) extends EndpointsWatch with Watch.Error[Endpoints]
-  }
+  case class Added(
+    `object`: Endpoints
+  ) extends EndpointsWatch with Watch.Added[Endpoints]
+
+  case class Modified(
+    `object`: Endpoints
+  ) extends EndpointsWatch with Watch.Modified[Endpoints]
+
+  case class Deleted(
+    `object`: Endpoints
+  ) extends EndpointsWatch with Watch.Deleted[Endpoints]
+
+  case class Error(
+    @JsonProperty(value = "object") status: Status
+  ) extends EndpointsWatch with Watch.Error[Endpoints]
```

Comment:
`v1` contains `implicit object EndpointsDescriptor` and implicit classes can't live in packages.  I tried splitting this up so that some things went into the package and others into the package object, but then hit another weird issue with an illegal classfile dependency between EndpointsWatch and its companion object.

Sooo... ¯\\\_(ツ)\_/¯